nixos/modules/services/web-apps/collabora-online.nix

```diff
@@ -0,0 +1,1638 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+
+let
+  cfg = config.services.collabora-online;
+
+  configFile =
+    let
+      c = cfg.config;
+      toStr =
+        x:
+        if x == null then
+          ""
+        else if lib.isBool x then
+          lib.boolToString x
+        else if lib.isInt x || lib.isFloat x then
+          toString x
+        else if lib.isString x then
+          lib.escapeXML x
+        else
+          throw "Unsupported type: ${builtins.toJSON x}";
+    in
+    # Example config file is taken from https://github.com/CollaboraOnline/online/blob/master/coolwsd.xml.in
+    pkgs.writeText "coolwsd.xml" ''
```

Comment:
How about we take the vendored file, import it from the package and then use an xml manipulation tool to substitute the values in?

Reply:
Rehauled in the RFC42-style.